lib/Frontend/CompilerInvocation.cpp

```diff
@@ -168,8 +168,11 @@ static void updateRuntimeLibraryPaths(SearchPathOptions &SearchPathOpts,
     LibSubDir = "maccatalyst";
 
   llvm::sys::path::append(LibPath, LibSubDir);
+  llvm::SmallString<128> runtimeLibPath = LibPath;
+  if (!Triple.isOSDarwin())
+    llvm::sys::path::append(runtimeLibPath, swift::getMajorArchitectureName(Triple));
   SearchPathOpts.RuntimeLibraryPaths.clear();
-  SearchPathOpts.RuntimeLibraryPaths.push_back(std::string(LibPath.str()));
+  SearchPathOpts.RuntimeLibraryPaths.push_back(std::string(runtimeLibPath.str()));
```

Comment:
Should this be in `if (!Triple.isOSDarwin())`? I would prefer to turn these lines into a single `if`, eg:
```cpp
if (!Triple.isOSDarwin()) {
  llvm::SmallString<128> runtimeLibPath = LibPath;
  llvm::sys::path::append(runtimeLibPath, swift::getMajorArchitectureName(Triple));
  SearchPathOpts.RuntimeLibraryPaths.push_back(std::string(runtimeLibPath.str()));
} else {
  SearchPathOpts.RuntimeLibraryPaths.push_back(DARWIN_OS_LIBRARY_PATH);
}
```

Reply:
Yes, this last line applies to Darwin, as there it simply does not add the architecture, so it is the same as LibPath before. Your code leaves out the `SearchPathOpts.RuntimeLibraryPaths.clear()`, which is needed because this function is run multiple times.

Reply:
```suggestion
  SearchPathOpts.RuntimeLibraryPaths.push_back(runtimeLibPath.str().str());
```

Reply:
@compnerd, just seeing this suggestion now, missed this when you wrote it a week back. Since this is changing unrelated behavior to my simple renaming, can we get your change in in a subsequent pull? I just don't want to start another hours-long CI run for this unrelated update.